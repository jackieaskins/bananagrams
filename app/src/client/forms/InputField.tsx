import { Input } from "@chakra-ui/react";
import FormField, { FormFieldProps } from "./FormField";

export interface InputFieldProps extends FormFieldProps {
  autoComplete?: string;
  setValue: (newValue: string) => void;
  type?: React.HTMLInputTypeAttribute;
  value: string;
}

export default function InputField({
  autoComplete,
  name,
  setValue,
  type,
  value,
  ...formFieldProps
}: InputFieldProps): JSX.Element {
  return (
    <FormField name={name} {...formFieldProps}>
      <Input
        autoComplete={autoComplete}
        id={name}
        type={type}
        value={value}
        onChange={(event) => setValue(event.target.value)}
        name={name}
      />
    </FormField>
  );
}
