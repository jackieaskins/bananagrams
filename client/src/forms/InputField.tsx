import React from 'react';
import Form from 'react-bootstrap/Form';
import { FormControlProps } from 'react-bootstrap/FormControl';

interface InputFieldProps extends FormControlProps {
  label?: React.ReactNode;
  name: string;
  setValue: (newValue: string) => void;
  value: string;
}

const InputField: React.FC<InputFieldProps> = ({
  label,
  name,
  setValue,
  value,
  ...rest
}) => (
  <Form.Group controlId={name}>
    {label ? <Form.Label>{label}</Form.Label> : null}
    <Form.Control
      {...rest}
      name={name}
      onChange={(event: React.ChangeEvent<HTMLInputElement>): void =>
        setValue(event.target.value)
      }
      value={value}
    />
  </Form.Group>
);

export default InputField;
