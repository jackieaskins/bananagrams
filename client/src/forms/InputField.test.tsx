import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import InputField from './InputField';

const mockSetValue = jest.fn().mockName('setValue');

describe('InputField', () => {
  const renderer = ShallowRenderer.createRenderer();

  const renderComponent = (propOverrides = {}) => {
    const props = {
      name: 'name',
      setValue: mockSetValue,
      value: 'value',
      ...propOverrides,
    };

    return renderer.render(<InputField {...props} />);
  };

  const getComponent = () => renderer.getRenderOutput();

  test('renders properly without label', () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  test('renders properly with label', () => {
    expect(renderComponent({ label: 'label' })).toMatchSnapshot();
  });

  test('onChange calls setValue', () => {
    const value = 'newValue';
    getComponent().props.children[1].props.onChange({ target: { value } });

    expect(mockSetValue).toHaveBeenCalledWith(value);
  });
});
