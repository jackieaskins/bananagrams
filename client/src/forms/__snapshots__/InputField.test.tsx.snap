// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField renders properly with label 1`] = `
<FormGroup
  controlId="name"
>
  <FormLabel
    column={false}
    srOnly={false}
  >
    label
  </FormLabel>
  <FormControl
    name="name"
    onChange={[Function]}
    value="value"
  />
</FormGroup>
`;

exports[`InputField renders properly without label 1`] = `
<FormGroup
  controlId="name"
>
  <FormControl
    name="name"
    onChange={[Function]}
    value="value"
  />
</FormGroup>
`;
